### DISCUSSION TITLE
SchemaField 的 props 属性 schema 的 properties 是一个对象，无法绝对保证属性顺序

### DISCUSSION THREAD

**Discussion by SignDawn** [alibaba/formily#3300] [Q&A 2.x] [answered]
查看源码得知，若不设置 x-index 属性，那么顺序则与 for in 提供的顺序相同。
![image](https://user-images.githubusercontent.com/48878568/182018505-436303d2-22d1-4ea0-849a-5a15a52fef03.png)
但对于 for in 而言，无法绝对保证对象的顺序，在正整数的 key 时，顺序不定。
![image](https://user-images.githubusercontent.com/48878568/182018692-7d95ae2c-29cc-49e3-86bf-72ff97e596fe.png)
当然，数字的场景应该也很少见，甚至说完全不必要，只是看到 properties 是一个对象比较疑惑顺序的问题。

**Comment by janryWang** [comment, marked as answer]
x-index是非常有必要的

**Reply by marspal** [reply]
x-index 相同时 能否顺序显示,  而不是直接忽略呢

**Comment by SignDawn** [comment]
如果用 x-index 是不是过于麻烦了？比如说现在有 1000 个表单项，现在要加其中一个，岂不是后面的 index 都要改？

**Reply by henryzp** [reply]
所以用平台生成schema，很有必要
